Model/Product/Cache/CacheRepository.php

```diff
@@ -62,17 +63,17 @@ class IndexRepository implements ProductIndexRepositoryInterface
     /**
      * IndexRepository constructor.
      *
-     * @param IndexResource $indexResource
-     * @param IndexCollectionFactory $indexCollectionFactory
+     * @param CacheResource $indexResource
```

Comment:
Maybe rename vars as well? If you do so, remember to check the DI.xml for the var name as well.